src/compiler/program.ts

```diff
@@ -2722,6 +2722,10 @@ namespace ts {
                 createDiagnosticForOptionName(Diagnostics.Option_paths_cannot_be_used_without_specifying_baseUrl_option, "paths");
             }
 
+            if (options.incremental && !options.configFilePath) {
+                programDiagnostics.add(createCompilerDiagnostic(Diagnostics.Option_incremental_can_only_be_specified_when_using_tsconfig));

```

Comment:
Why does this restriction exist? Is it to determine the name of the `.tsbuildinfo` file based on the path of the tsconfig.json?
That would make it impossible for API users to use the `incremental` option if they don't use `parseJsonConfigFileContent` (or one of the countless similar functions) to create their `compilerOptions` because `configFilePath` is typically not set.

Would it make sense to allow `incremental` if either `configFilePath` or `tsBuildInfoFile` is defined?

Reply:
Agree - would be very nice to just specify a metadata location and then not need a tsconfig. Actually, we probably _need_ that capability to support `--incremental` in Visual Studio.

Reply:
@RyanCavanaugh @DanielRosenwasser What do you think. In that case we would want to make `tsBuildInfoFile` as command line argument.

Reply:
The release announcement made it look like there was already a `--tsBuildInfoFile` CLI option